dagql/idproto/id.go

```diff
@@ -202,20 +202,32 @@ func (id *ID) Canonical() *ID {
 	return canon
 }
 
-var digests = new(sync.Map)
+var digestCache *sync.Map
+
+// EnableDigestCache enables caching of digests for IDs.
+//
+// This is not thread-safe and should be called before any IDs are created, or
+// at least digested.
+func EnableDigestCache() {
```

Comment:
So this only affects the new `dagger call` - I think we should still follow-up and try and clean this cache once it gets enabled by default for `dagger run` as well.